src/pykeen/nn/node_piece/representations.py

```diff
@@ -298,3 +315,43 @@ def _plain_forward(
             ),
             self.aggregation_index,
         )
+
+    def ratio_unique_hashes(self) -> RatioInfo:
+        """
+        Return the ratio of unique hashes in the total pool.
+
+        :return:
+            A ratio information tuple
+
+        .. todo:: @migalkin explain why you would want to calculate this
+
+        Example usage:
+
+        .. code-block::
+
+            from pykeen.model import NodePiece
+
+            model = NodePiece(
+                triples_factory=dataset.training,
+                tokenizers=["AnchorTokenizer", "RelationTokenizer"],
+                num_tokens=[20, 12],
+                embedding_dim=64,
+                interaction="rotate",
+                relation_constrainer="complex_normalize",
+                entity_initializer="xavier_uniform_",
+            )
+            print(model.entity_representations[0].num_unique_hashes())
+        """
+        # unique hashes per representation
+        uniques_per_representation = [
+            tokens.assignment.unique(dim=0).shape[0] / self.max_id for tokens in self.token_representations
+        ]
+
+        # unique hashes if we concat all representations together
+        uniques_total = torch.unique(
+            torch.cat([tokens.assignment for tokens in self.token_representations], dim=-1), dim=0
+        ).shape[0]
+        return RatioInfo(
+            unique_per_repr=uniques_per_representation,
```

Comment:
can we improve the names of the variables in this function as well as the RatioInfo object?

Reply:
my varname creativity is on the dry side those days, I'm fine with the current ones 😅

Reply:
two specific things:

1. mismatch in `unique_per_repr` and `uniques_per_representation` for the first one -> just pic one
2. `unique_total=uniques_total / self.max_id` is confusing because the variable `uniques_total` seems to be normalized by `max_id`, maybe better to call `unnormalized_uniques_total`

Reply:
Done in 7afa1b8b :)